yt_dlp/extractor/googledrive.py

```diff
@@ -166,6 +179,10 @@ def _get_automatic_captions(self, video_id, subtitles_id, hl):
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        _, webpage_urlh = self._download_webpage_handle(url, video_id)
+        if webpage_urlh.url != url:
+            return self.url_result(webpage_urlh.url)
```

Comment:
maybe it'd be better to just update the url(and maybe also the video id?), and raise login required when necessary